ml-agents/mlagents/trainers/learn.py

```diff
@@ -20,36 +27,145 @@ def run_training(sub_id, run_seed, run_options, process_queue):
-        if run_options['--curriculum'] != 'None' else None)
+    curriculum_folder = (run_options['--curriculum']
+                         if run_options['--curriculum'] != 'None' else None)
     lesson = int(run_options['--lesson'])
     fast_simulation = not bool(run_options['--slow'])
     no_graphics = run_options['--no-graphics']
     trainer_config_path = run_options['<trainer-config-path>']
 
-    # Create controller and launch environment.
-    tc = TrainerController(env_path, run_id + '-' + str(sub_id),
-                           save_freq, curriculum_file, fast_simulation,
-                           load_model, train_model, worker_id + sub_id,
-                           keep_checkpoints, lesson, run_seed,
-                           docker_target_name, trainer_config_path, no_graphics)
+    # Recognize and use docker volume if one is passed as an argument
+    if not docker_target_name:
+        model_path = './models/{run_id}'.format(run_id=run_id)
+        summaries_dir = './summaries'
+    else:
+        trainer_config_path = \
+            '/{docker_target_name}/{trainer_config_path}'.format(
+                docker_target_name=docker_target_name,
+                trainer_config_path=trainer_config_path)
+        if curriculum_folder is not None:
+            curriculum_folder = \
+                '/{docker_target_name}/{curriculum_folder}'.format(
+                    docker_target_name=docker_target_name,
+                    curriculum_folder=curriculum_folder)
+        model_path = '/{docker_target_name}/models/{run_id}'.format(
+            docker_target_name=docker_target_name,
+            run_id=run_id)
+        summaries_dir = '/{docker_target_name}/summaries'.format(
+            docker_target_name=docker_target_name)
+
+    env = init_environment(env_path, docker_target_name, no_graphics, worker_id + sub_id, fast_simulation, run_seed)
+
+    if curriculum_folder is None:
+        meta_curriculum = None
+    else:
+        meta_curriculum = MetaCurriculum(curriculum_folder, env._resetParameters)
+
+    if meta_curriculum:
+        for brain_name in meta_curriculum.brains_to_curriculums.keys():
+            if brain_name not in env.external_brain_names:
+                raise MetaCurriculumError('One of the curricula '
+                                          'defined in ' +
+                                          curriculum_folder + ' '
+                                          'does not have a corresponding '
+                                          'Brain. Check that the '
+                                          'curriculum file has the same '
+                                          'name as the Brain '
+                                          'whose curriculum it defines.')
+
+    external_brains = {}
+    for brain_name in env.external_brain_names:
+        external_brains[brain_name] = env.brains[brain_name]
+    trainer_config = load_config(trainer_config_path)
```

Comment:
I think this should be called before the environment is instantiated. This way if there is an error when loading the config, we can raise it before opening the environment. 
_Note this might be out of the scope of this PR_

Reply:
This seems relevant and small enough for me to do for this PR, will fix.